File: hiddenwheelviewlibrary/src/main/java/me/tictok/hiddenwheelviewlibrary/CustomSnapHelper.java
package me.tictok.hiddenwheelviewlibrary;

import android.support.v7.widget.LinearSnapHelper;
import android.support.v7.widget.RecyclerView;
import android.view.View;

public class CustomSnapHelper extends LinearSnapHelper {

    public int getSnapPosition(RecyclerView recyclerView) {
        RecyclerView.LayoutManager layoutManager = recyclerView.getLayoutManager();
        if (layoutManager == null) return RecyclerView.NO_POSITION;
        View snapView = findSnapView(layoutManager);
        if (snapView == null) return RecyclerView.NO_POSITION;
        return layoutManager.getPosition(snapView);
    }
}

File: library/src/main/java/me/tictok/library/CustomSnapHelper.kt
package me.tictok.library

import android.support.v7.widget.LinearSnapHelper
import android.support.v7.widget.RecyclerView

class CustomSnapHelper : LinearSnapHelper() {

    fun getSnapPosition(recyclerView: RecyclerView): Int {
        val layoutManager = recyclerView.layoutManager ?: return RecyclerView.NO_POSITION
        val snapView = findSnapView(layoutManager) ?: return RecyclerView.NO_POSITION
        return layoutManager.getPosition(snapView)
    }
}
